packages/rest/test/integration/http-handler.integration.ts

```diff
@@ -72,11 +78,19 @@ describe('HttpHandler', () => {
         public async postHello(): Promise<string> {
           return 'hello posted';
         }
+
+        public async createHello(): Promise<string> {
+          return 'hello created';
```

Comment:
We should be able to allow a method to return an instance of `HttpResponse` or `Promise<HttpResponse>`, for example:

```ts
return HttpResponse.statusCode(201).header('Location', '...').body(...);
```

Reply:
I have a similar proposal in https://github.com/strongloop/loopback-next/issues/788#issuecomment-381639062:

```ts
@post('/products')
  async create(@requestBody() obj: Product)
    : Promise<ControllerResponse<Product>> {
    const model = await this.productRepository.create(obj);
    return {
      statusCode: 201,
      // FIXME: Do not hard-code `model.id`, find the actual PK property name.
      // FIXME: Can Location contain a path only, with no hostname and scheme?
      headers: {location: `/products/${model.id}`},
      body: model,
    }
  }
```

IIUC, @jannyHou is exploring a different approach in this spike pull request, one where the status code is automatically inferred from the OpenAPI spec.